src/design-pages/passkey_authentication.rst

```diff
@@ -0,0 +1,313 @@
+like big companies and governments, where the usage of these authentication
+mechanisms is becoming a common pattern.
+
+SSSD should provide a way to authenticate users that use a passkey. This would
+add more value by attracting the types of users mentioned in the previous
+paragraph.
+
+For the purpose of this feature, passkey is a FIDO2 compatible device supported
+by the libfido2 library.
+
+Use cases
+---------
+As a user, I want to walk to any system on a network and log in (desktop and/or
+terminal) using a passkey device connected to the system. The user credentials
+will be stored in a central management system like FreeIPA or Active Directory.
+
+Solution overview
+------------------------
+The objective is to use a passkey to locally authenticate a user against 
+centralized identity management system. For that purpose an integration with an
+LDAP server like AD or FreeIPA is needed. SSSD should get the user credentials
+from this server, including the user key mapping data. On top of that, SSSD
+should communicate with the key to get the information stored in the device.
+Once this is done, SSSD should use all this data to authenticate the user.
+
+The biggest advantage of this approach is the usage of a central repository to
+store the user's credentials. In the long term this gives more flexibility as
+the users and keys can be authorized and revoked by the system administrator
+easily.
+
+Existing solution comparison
+****************************
+Existing Yubikey devices have already been enabled in RHEL and Fedora but the
+approach is completely different. The existing solution uses
+`OATH-TOTP <https://en.wikipedia.org/wiki/Time-based_one-time_password>`__ and
+`OATH-HOTP <https://en.wikipedia.org/wiki/HMAC-based_one-time_password>`__ to
+generate a one-time password (OTP) that is sent to the server as a second
+factor authentication. In other words, the authentication checks happen on the
+server. Moreover, this solution relies on a plugin for the 389 directory
+server, so it only works with IPA for IPA users.
+
+The approach explained in this document relies on the 
+`FIDO2 <https://en.wikipedia.org/wiki/FIDO2_Project>`__ standard, which uses a
+public-private key scheme to perform the authentication checks on the client.
+Besides, the solution is general enough to accept different LDAP servers like
+AD and IPA.
+
+Another difference is that with the current IPA feature, the first and second
+factor can also be used for doing a direct LDAP bind. This will not be possible
+with the new passkey scheme. Direct LDAP bind support will not be in the scope
+of this implementation.
+
+Implementation details
+----------------------
+
+The implementation will loosely follow existing smartcard authentication in
+SSSD. On top of that, and as mentioned before, the information will be stored
+in two places: in the LDAP server and in the passkey. The server will contain
```

Comment:
s/in the passkey/on the passkey/  sounds a bit better IMO

Reply:
Yeah, this is a recurrent problem for Spanish-speaking, we fail when using in/on/at. Thanks for catching it.